### PR TITLE
python3.pkgs.dvc, dvc, dvc-with-remotes: move to python-modules since dvclive depends on it

### PR DESCRIPTION
**Pull request by tjni**
## Description of changes

This follows the best practices in the documentation.

There's also a practical reason to do this. I was confused by my nixpkgs-review results on https://github.com/NixOS/nixpkgs/pull/246988, which only displayed a cryptic error:

```
Executing pipInstallPhase
/private/tmp/nix-build-python3.11-dvclive-2.13.0.drv-0/source/dist /private/tmp/nix-build-python3.11-dvclive-2.13.0.drv-0/source
Processing ./dvclive-2.13.0-py3-none-any.whl
ERROR: Could not find a version that satisfies the requirement dvc>=2.58.0 (from dvclive) (from versions: none)
ERROR: No matching distribution found for dvc>=2.58.0

error: builder for '/nix/store/0lzhxchgywz4qz3qqa200xvjl7cry1j0-python3.11-dvclive-2.13.0.drv' failed with exit code 1;
       last 10 log lines:
       > Successfully built dvclive
       > Finished creating a wheel...
       > Finished executing pipBuildPhase
       > installing
       > Executing pipInstallPhase
       > /private/tmp/nix-build-python3.11-dvclive-2.13.0.drv-0/source/dist /private/tmp/nix-build-python3.11-dvclive-2.13.0.drv-0/source
       > Processing ./dvclive-2.13.0-py3-none-any.whl
       > ERROR: Could not find a version that satisfies the requirement dvc>=2.58.0 (from dvclive) (from versions: none)
       > ERROR: No matching distribution found for dvc>=2.58.0
       >
       For full logs, run 'nix-store -l /nix/store/0lzhxchgywz4qz3qqa200xvjl7cry1j0-python3.11-dvclive-2.13.0.drv'.
```

Moving `dvc` to python-modules reveals the reason clearly:

```
❯ NIXPKGS_ALLOW_BROKEN=1 nix-build -A python311.pkgs.dvclive --argstr --system aarch64-linux                                           took 23s
error:
       ...

       error: pyro4-4.82 not supported for interpreter python3.11
```

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/dvc/default.nix [hunk 1]
```diff
@@ -1,14 +1,59 @@
...
+, tabulate
+, tomlkit
+, tqdm
+, typing-extensions
+, voluptuous
+, zc_lockfile
 , enableGoogle ? false
 , enableAWS ? false
 , enableAzure ? false
 , enableSSH ? false
```
Comment (natsukium) [inline pkgs/development/python-modules/dvc/default.nix:53]: We can leave these for applications, but how about setting optional-dependencies for libraries?

Comment (tjni) [reply pkgs/development/python-modules/dvc/default.nix:53]: You're absolutely right, I missed that. Let me update it some more.

Comment (tjni) [reply pkgs/development/python-modules/dvc/default.nix:53]: This should be fixed now.

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 247019` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>dvc (python310Packages.dvc)</li>
    <li>dvc-with-remotes</li>
    <li>dvc-with-remotes.dist</li>
    <li>dvc.dist (python310Packages.dvc.dist)</li>
    <li>python310Packages.dvclive</li>
    <li>python310Packages.dvclive.dist</li>
  </ul>
</details>

**Comment by tjni** [comment]
Those packages build on aarch64-linux for me. x86_64-linux is the only one I don't know how to run on my Mac. Would you mind pasting some of the errors you saw?

**Review by natsukium — APPROVED** [review]
It looks like the `python310Packages.moto` that depends is broken on HEAD.
This PR change itself does not seem to be a problem.

**Comment by tjni** [comment]
> It looks like the python310Packages.moto that depends is broken on HEAD.

That's too bad :(

Thank you taking the time to build and review this.

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 247019` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>dvc (python310Packages.dvc)</li>
    <li>dvc-with-remotes</li>
    <li>dvc-with-remotes.dist</li>
    <li>dvc.dist (python310Packages.dvc.dist)</li>
    <li>python310Packages.dvclive</li>
    <li>python310Packages.dvclive.dist</li>
  </ul>
</details>

**Pull request merged**